python/ribasim/ribasim/config.py

```diff
@@ -82,6 +82,8 @@ class Terminal(NodeModel):
         default_factory=TableModel[TerminalStaticSchema]
     )
 
+    _sort_keys: dict[str, list[str]] = {"static": ["node_id"]}
```

Comment:
I don't agree with adding this protected variable in multiple places. There are other ways to do this. For example, add an `__init__` function everywhere to add the required keys to the dict.

But I am trying to look a bit further. It would be better if we receive this information from the julia code. For example, by adding it in the schemas and reading the schemas in python. Would that be something that is possible with pydantic?

Reply:
`_sort_keys` was already used, falling back to a wrong default if it wasn't present. This PR removes the wrong default and requires explicitly stating the correct sorting order.

If possible I agree it would be nice to share this or run this through pydantic. I'd like to keep this PR focused on a bugfix and not add a refactor on top.

Reply:
@visr @evetion take a look at feat/sort-keys-in-tablemodel. I don't have it working yet. But to me it seems like we need to put the sort keys inside of the table model as some class var which depends per initialization. Or maybe in another way. Just a Field?

Reply:
I have been playing a bit more with this, and it seems to me that the best option would be to make the sort_keys an instance Field

```py
sort_keys: list[str] = Field(init_var=True, exclude=True, repr=False)
```

I set it to init_var, so it can never be changed afterwards, and I set it to exclude, because the data is always set in python.

The only problem that I still had was the moment that you set a pandas DataFrame, because then it tells me that:

> pydantic_core._pydantic_core.ValidationError: 1 validation error for Terminal
static.sort_keys
  Input should be a valid list [type=list_type, input_value=None, input_type=NoneType]        
    For further information visit https://errors.pydantic.dev/2.5/v/list_type
    
I believe we could solve this by using a different scheme for custom data types, instead of the `@model_validator` as described here: https://docs.pydantic.dev/latest/concepts/types/#custom-types

Reply:
Is init_var=True needed? Because if you set a DataFrame, the TableModel already exists, including its specific sort_keys. Ideally we get that from somewhere.

> But I am trying to look a bit further. It would be better if we receive this information from the julia code. For example, by adding it in the schemas and reading the schemas in python. Would that be something that is possible with pydantic?

Fully agree, but I couldn't find a way of doing so. It requires some custom info passed to the JSON Schema, the https://github.com/koxudaxi/datamodel-code-generator/, to read it and pass it into some class as a private field, or Config option, that we here can pick up on.

Reply:
I created #956 for the first effort on getting the sort_keys in pydantic. You could decide to enhance the ticket, or otherwise just make a new ticket to get the info from Julia.